pkg/expr/eval_test.go

```diff
@@ -0,0 +1,375 @@
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package expr
+
+import (
+	"strings"
+	"testing"
+	"time"
+
+	"istio.io/mixer/pkg/attribute"
+)
+
+func TestGoodEval(t *testing.T) {
+	tests := []struct {
+		src    string
+		tmap   map[string]interface{}
+		result interface{}
+		err    string
+	}{
+		{
+			"a == 2",
+			map[string]interface{}{
+				"a": int64(2),
+			},
+			true, "",
+		},
+		{
+			"a != 2",
+			map[string]interface{}{
+				"a": int64(2),
+			},
+			false, "",
+		},
+		{
+			"a != 2",
+			map[string]interface{}{
+				"d": int64(2),
+			},
+			false, "unresolved attribute",
+		},
+		{
+			"a ",
+			map[string]interface{}{
+				"a": int64(2),
+			},
+			int64(2), "",
+		},
+		{
+			"2 ",
+			map[string]interface{}{
+				"a": int64(2),
+			},
+			int64(2), "",
+		},
+		{
+			`request.user == "user1"`,
+			map[string]interface{}{
+				"request.user": "user1",
+			},
+			true, "",
+		},
+		{
+			`request.user2| request.user | "user1"`,
+			map[string]interface{}{
+				"request.user": "user2",
+			},
+			"user2", "",
+		},
+		{
+			`request.user2| request.user3 | "user1"`,
+			map[string]interface{}{
+				"request.user": "user2",
+			},
+			"user1", "",
+		},
+		{
+			`request.size| 200`,
+			map[string]interface{}{
+				"request.size": int64(120),
+			},
+			int64(120), "",
+		},
+		{
+			`request.size| 200`,
+			map[string]interface{}{
+				"request.size": int64(0),
+			},
+			int64(200), "",
+		},
+		{
+			`request.size| 200`,
+			map[string]interface{}{
+				"request.size1": int64(0),
+			},
+			int64(200), "",
+		},
+		{
+			`(x == 20 && y == 10) || x == 30`,
+			map[string]interface{}{
+				"x": int64(20),
+				"y": int64(10),
+			},
+			true, "",
+		},
+		{
+			`service.name == "*.ns1.cluster" && service.user == "admin"`,
+			map[string]interface{}{
+				"service.name": "svc1.ns1.cluster",
+				"service.user": "admin",
+			},
+			true, "",
+		},
+		{
+			`(x/y) == 30`,
+			map[string]interface{}{
+				"x": int64(20),
+				"y": int64(10),
+			},
+			false, "unknown function: QUO",
+		},
+		// The following will not work until we support
+		// map as an attribute type
+		{
+			`request.header["X-FORWARDED-HOST"] == "aaa"`,
+			map[string]interface{}{
+				"request.header": map[string]string{
+					"X-FORWARDED-HOST": "bbb",
+				},
+				"y": int64(10),
+			},
+			true, "unresolved attribute",
+		},
+	}
+
+	for idx, tst := range tests {
+		attrs := &bag{attrs: tst.tmap}
+		exp, err := Parse(tst.src)
+		if err != nil {
+			t.Errorf("[%d] unexpected error: %s", idx, err)
```

Comment:
just as an FYI: I don't see a change here to the tests. not a blocker, but I'm wondering if there is a missing commit ?